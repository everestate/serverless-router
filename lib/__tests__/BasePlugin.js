const BasePlugin = require('../BasePlugin');


describe('BasePlugin', () => {
  describe('countructor(routes)', () =>
    test('initializes .routes property', () => {
      const routes = ['foo', 'bar'];
      const plugin = new BasePlugin(routes);
      expect(plugin.routes).toEqual(routes);
    }));

  describe('appendRoute(...args)', () => {
    const origMatchFn = BasePlugin.match;
    beforeEach(() => { BasePlugin.match = jest.fn(() => 42); });
    afterEach(() => { BasePlugin.match = origMatchFn; });

    test('appends matching route', () => {
      const plugin = new BasePlugin([]);
      plugin.appendRoute('a', 'b', 'c', 'd', 'e', 'f');
      expect(plugin.routes).toEqual([{
        callback: 'f',
        match: 42,
        pluginName: 'baseplugin',
      }]);
      expect(BasePlugin.match).toHaveBeenCalledTimes(1);
      expect(BasePlugin.match).toHaveBeenCalledWith('a', 'b', 'c', 'd', 'e');
    });

    test('responds with instance of the plugin', () => {
      const plugin = new BasePlugin([]);
      expect(plugin.appendRoute()).toEqual(plugin);
    });
  });

  describe('.pluginName', () =>
    test('responds with normalized plugin name', () =>
      expect((class MyPlugin extends BasePlugin {}).pluginName).toEqual('myplugin')));

  describe('static match(subject, callback)', () =>
    test('throws "Not implemented" exception', () =>
      expect(BasePlugin.match).toThrowErrorMatchingSnapshot()));
});
