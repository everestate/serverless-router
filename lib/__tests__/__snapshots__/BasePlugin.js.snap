// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasePlugin static match(subject, callback) throws "Not implemented" exception 1`] = `"Not implemented"`;
